editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5973,9 +6031,14 @@ void CanvasItemEditorViewport::_perform_drop_data() {
 		Ref<Texture2D> texture = res;
 		if (texture.is_valid()) {
 			Node *child = Object::cast_to<Node>(ClassDB::instantiate(default_texture_node_type));
-			_create_nodes(target_node, child, path, drop_pos);
+			_create_texture_node(target_node, child, path, drop_pos);
 			undo_redo->add_do_method(editor_selection, "add_node", child);
 		}
+
+		Ref<AudioStream> audio = res;
+		if (audio.is_valid()) {
+			_create_audio_node(target_node, path, drop_pos);
```

Comment:
I don't get why this method takes path instead of the resource it just loaded. (I know it's the same as above, I just wonder)

Reply:
The name of the node is taken from the path of the resource. Although it is possible to get from `Resource.resource_path`.

Reply:
Yep.